snap-gpf/src/main/java/org/esa/snap/core/gpf/common/ReadOp.java

```diff
@@ -115,7 +118,7 @@ public void initialize() throws OperatorException {
         boolean hasBandNames = (this.bandNames != null && this.bandNames.length > 0);
         boolean hasMaskNames = (this.maskNames != null && this.maskNames.length > 0);
         ProductSubsetDef subsetDef = null;
-        if (hasBandNames || hasMaskNames || this.pixelRegion != null || this.geometryRegion != null || this.copyMetadata) {
+        if (hasBandNames || hasMaskNames || this.pixelRegion != null || this.geometryRegion != null || !this.copyMetadata) {
```

Comment:
I think here the copyMetadata should not be inverted, right?

Reply:
I inverted the if condition for the 'copyMetadata' attribute (from if (copyMetadata){...} to if (!copyMetadata){...}) to avoid creating allways the ProductSubsetDef object only for metadata when no band names, no mask names, no region are specified.

From my point of view if something (a graph, an operator) invokes the ReadOp operator just  for reading the source product with all its bands, masks, metadata without specifying
certain bands, masks, a region to read, the subset def object should not be cretead.

Reply:
Yes, this is true.